### PR TITLE
Changed virtual/pure virtual

### PR DESCRIPTION
**Pull request by jordiluong**
Changed some virtual to pure virtual and vice versa to make it more logical for implementing the skills

### CODE REVIEW

File: include/roboteam_ai/stp/Skill.h [hunk 1]
```diff
@@ -56,20 +56,20 @@ namespace rtt::ai::stp {
          * Terminates the skill
          * @return Status of termination
          */
-        virtual Status onTerminate() noexcept;
+        virtual Status onTerminate() noexcept {};
```
Comment (RobotJesse) [inline include/roboteam_ai/stp/Skill.h]: should be purely virtual

Comment (RobotJente) [inline include/roboteam_ai/stp/Skill.h]: I think this should be purely virtual
```suggestion
        virtual Status onTerminate() noexcept = 0;
```

File: include/roboteam_ai/stp/Skill.h [hunk 2]
```diff
@@ -56,20 +56,20 @@ namespace rtt::ai::stp {
...
          */
-        virtual Status onUpdate(SkillInfo const& info) noexcept;
+        virtual Status onUpdate(SkillInfo const& info) noexcept = 0;
 
         /**
          * Initializes the skill
          * @return Status of initialization
          */
-        virtual Status onInitialize() noexcept;
+        virtual Status onInitialize() noexcept {};
```
Comment (RobotJente) [inline include/roboteam_ai/stp/Skill.h]: This too 
```suggestion
        virtual Status onInitialize() noexcept = 0;
```

File: include/roboteam_ai/stp/Skill.h (hunk 1 above)
Comment (jordiluong) [reply include/roboteam_ai/stp/Skill.h]: I was thinking about this. It is cleaner to make it pure, so it is the responsibility of the derived skills. But the skills mostly do not need termination/initialization, so having it not pure is just convenience. If you think making it pure would be better, I'll go with that.

**Review by RobotJente — APPROVED** [review]
looks good

**Pull request merged**